modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ComponentMessageProcessorObjectFactory.java

```diff
@@ -82,7 +88,13 @@ public P doGetObject() {
                                                                              nestedChain)));
 
       // For MULE-18771 we need access to the chain's location to create a new event and sdk context
-      nestedChain.setAnnotations(this.getAnnotations());
+      // Update: For W-15158118, to avoid issues with the registration of the chain in the component locator, we are not
```

Comment:
explain that the issue is that because both the chain and scope were sharing the same location, the chain overrides the scope in the Locator class